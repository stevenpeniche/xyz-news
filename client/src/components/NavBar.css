.nav-bar {
  display: flex;
  position: fixed;
  align-items: center;
  width: 100%;
  height: 38px;
  overflow: hidden;
  background-color: #000000;
  border-bottom-style: solid;
  border-bottom-width: 1px;
  border-color: hsla(0, 0%, 0%, 0.1);
  box-shadow: 0 1px 2px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23);
  z-index: 1;
}

#logo {
  order: 1;
  margin-left: 20px;
  margin-right: 20px;
  font-family: 'Audiowide', sans-serif;
  font-size: 30px;
  letter-spacing: -2px;
  color: #ffffff;
  text-decoration: none;
}

.nav-bar-menu {
  order: 2;
}

.nav-bar-menu-icon {
  margin-top: 5px;
  font-size: 20px;
  color: #ffffff;
  text-decoration: none;
  animation-delay: .4s;
}

.nav-bar-menu-icon:hover {
  cursor: pointer;
  color: #ffffffb7;
}

.nav-bar-menu-list-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  position: fixed;
  top: 10px;
  left: 220px;
  z-index: 2;
  min-width: 120px;
  background-color: #ffffff;
  -moz-border-radius: 8px;
	-webkit-border-radius: 8px;
	border-radius: 8px;
  box-shadow: 0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23);
  overflow: hidden;
  animation-duration: .4s;
}

.nav-bar-menu-list {
  margin-bottom: 10px;
}

.nav-bar-menu-list-item {
  text-align: center;
  text-decoration: none;
  color: #000000;
}

.menu-item-source-icon-container {
  margin-top: 10px;
}

.scroll-to-top-section {
  order: 3;
  flex-grow: 1;
  height: 100%;
  margin-left: 20px;
}

.scroll-to-top-section:hover {
  cursor: pointer;
}

