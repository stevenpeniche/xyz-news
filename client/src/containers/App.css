.source-icon-container {
  display: flex;
  justify-content: center;
}

.source-icon {
  width: 60px;
  height: 60px;
	-moz-border-radius: 50px;
	-webkit-border-radius: 50px;
  border-radius: 50px;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
  transition: all 0.3s cubic-bezier(.25,.8,.25,1);
  animation-delay: .5s;
}



.levitate:hover {
  box-shadow: 0 14px 28px rgba(0,0,0,0.25), 0 10px 10px rgba(0,0,0,0.22);
}