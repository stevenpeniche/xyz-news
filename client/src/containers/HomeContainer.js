import React, { Component } from 'react';
import Home from '../components/Home';

class HomeContainer extends Component {
  componentDidMount() {
    this.props.changeCurrentSource();
  }

  render() {
    return (
      <Home sources={this.props.sources}/>
    );
  }
}

export default HomeContainer;