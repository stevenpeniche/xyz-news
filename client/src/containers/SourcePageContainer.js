import React, { Component } from 'react';
import axios from 'axios';
import SourcePage from '../components/SourcePage';

class SourcePageContainer extends Component  {
  constructor(props){
    super(props);
    this.changeCurrentTopic = this.changeCurrentTopic.bind(this);
    this.state = {
      currentTopic: 'Top',
      topicData: []
    };
  }

  changeCurrentTopic(topic) {
    const source = this.props.source

    if(topic !== this.state.currentTopic) {
      const apiPath =  topic === 'Top' ?
                      `/newsapi?source=${source.id}` :
                      `/newsapi?source=${source.id}&q=${topic}`;

      this.callAPI(apiPath)
          .then(res => {
            this.setState({
              currentTopic: topic,
              topicData: res.data
            });
          })
          .catch(err => console.log(err));

      this.setState({
        currentTopic: topic
      });
    }
  };

  callAPI = async (path) => {
    const response = await axios.get(path);
    const body = await response;

    return body;
  }

  componentDidMount() {
    const source = this.props.source
    this.props.changeCurrentSource(source.id);

    this.callAPI(`/newsapi?source=${source.id}`)
        .then(res => {
          this.setState({
            topicData: res.data
          });
        })
        .catch(err => console.log(err));
  }

  render() {
    return (
      <SourcePage
        className="source-page-component"
        source={this.props.source}
        currentTopic={this.state.currentTopic}
        changeCurrentTopic={this.changeCurrentTopic}
        topicData={this.state.topicData}
      />
    );
  }
}

export default SourcePageContainer;