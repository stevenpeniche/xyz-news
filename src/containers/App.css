.app {
  height: 100vh;
}

.source {
  display: inline-block;
  text-decoration: none;
  color: #000000;
  font-family: 'Roboto', sans-serif;
}

.source-icon-container {
  display: flex;
  justify-content: center;
}

.source-icon {
  width: 60px;
  height: 60px;
	background: #ff6600;
	-moz-border-radius: 50px;
	-webkit-border-radius: 50px;
  border-radius: 50px;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
  transition: all 0.3s cubic-bezier(.25,.8,.25,1);
}

.source-title {
  margin-top: 5px;
  font-size: 14px;
}

.leviate:hover {
  box-shadow: 0 14px 28px rgba(0,0,0,0.25), 0 10px 10px rgba(0,0,0,0.22);
}