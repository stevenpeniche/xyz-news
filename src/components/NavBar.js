import React from 'react';
import { Link } from 'react-router-dom';
import './NavBar.css';

const NavBar = (props) => {
  return (
    <nav className="nav-bar">
      <Link to="/" id="logo">XYZ news</Link>
      {props.currentSource ?
        <Link to="#" className="nav-bar-link">Sources<i className="down-icon fas fa-chevron-down"></i></Link>
        :
        <div></div>
      }
    </nav>
  );
}

export default NavBar;