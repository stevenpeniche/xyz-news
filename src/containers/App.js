import React, { Component } from 'react';
import { BrowserRouter } from 'react-router-dom';
import HomeContainer from './HomeContainer';

class App extends Component {
  render() {
    return (
      <BrowserRouter>
        <div className="App">
          <HomeContainer />
        </div>
      </BrowserRouter>
    );
  }
}

export default App;
