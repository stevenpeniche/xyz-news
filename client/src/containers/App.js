import React, { Component } from 'react';
import { BrowserRouter, Route } from 'react-router-dom';
import NavBar from '../components/NavBar';
import HomeContainer from './HomeContainer';
import SourcePageContainer from './SourcePageContainer';
import './App.css';

const faviconFinder = (rootURL) => {
  return `https://besticon-demo.herokuapp.com/icon?url=${rootURL}/&size=70..120..200`
}

class App extends Component {
  constructor(props) {
    super(props);
    this.changeCurrentSource = this.changeCurrentSource.bind(this);
    this.state = {
      sources: [
        {
          id: 'hacker-news',
          title: 'Hacker News',
          faviconURL: faviconFinder('https://news.ycombinator.com'),
          primaryColor: '#ff6600',
          topics: [
            {
              name: 'Top'
            },
            {
              name: 'New'
            },
            {
              name: 'Ask'
            },
            {
              name: 'Jobs'
            }
          ]
        },
        {
          id: 'bleacher-report',
          title: 'Bleacher Report',
          faviconURL: faviconFinder('https://bleacherreport.com'),
          primaryColor: '#000000',
          topics: [
            {
              name: 'Top'
            },
            {
              name: 'MLB'
            },
            {
              name: 'NFL'
            },
            {
              name: 'NBA'
            },
            {
              name: 'UFC'
            },
            {
              name: 'Boxing'
            },
            {
              name: 'WWE'
            }
          ]
        }
      ],
      currentSourceID: null
    }
  };

  changeCurrentSource(source=null) {
    this.setState({
      currentSourceID: source
    });
  };

  render() {
    return (
      <BrowserRouter>
        <div className="app">
          <NavBar currentSourceID={this.state.currentSourceID}/>
          <Route
            exact path="/"
            render={() => (
              <HomeContainer
                sources={this.state.sources}
                changeCurrentSource={this.changeCurrentSource}
              />
            )}
          />
          {
            this.state.sources.map((source) => {
              return(
                <Route
                  key={source.id}
                  exact path={`/${source.id}`}
                  render={(props) => (
                    <SourcePageContainer
                      source={source}
                      path={props.match.path}
                      changeCurrentSource={this.changeCurrentSource}
                    />
                  )}
                />
              )
            })
          }
        </div>
      </BrowserRouter>
    );
  };
}

export default App;
