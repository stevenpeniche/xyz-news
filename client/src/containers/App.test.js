import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import App from './App';
import { BrowserRouter, Route } from 'react-router-dom';
import NavBar from '../components/NavBar';
import HomeContainer from './HomeContainer';
import SourcePageContainer from './SourcePageContainer';

const wrapper = shallow(<App />)

describe('App rendering', () => {
  it('renders without crashing', () => {
    wrapper
  });

  it('renders correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders the NavBar once", () => {
    expect(wrapper.find(NavBar).length).toEqual(1);
  });

  it("renders the BrowserRouter once", () => {
    expect(wrapper.find(BrowserRouter).length).toEqual(1);
  });

  it("renders a route for the HomeContainer", () => {
    expect(wrapper.find('.home-route').length).toEqual(1);
  });

  it("renders a route for every source available", () => {
    expect(wrapper.find('.source-page-route').length)
      .toEqual(wrapper.state().sources.length);
  });
});

describe("App initial state", () => {
  const state = wrapper.state();
  it("sets initial state of currentSourceID to null", () => {
    expect(state.currentSourceID).toBe(null);
  })
});

