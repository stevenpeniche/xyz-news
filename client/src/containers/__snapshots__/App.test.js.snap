// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App rendering renders correctly 1`] = `
<div
  className="app"
>
  <nav
    className="nav-bar"
  >
    <a
      href="/"
      id="logo"
      onClick={[Function]}
    >
      XYZ news
    </a>
    <div />
    <div
      className="scroll-to-top-section"
      onClick={[Function]}
    >
      ...
    </div>
  </nav>
  <div
    className="home"
  >
    <div
      className="sources animated fadeInDown"
    >
      <a
        className="source animated fadeIn"
        href="/bleacher-report"
        onClick={[Function]}
      >
        <div
          className="source-icon-container"
        >
          <img
            alt="Bleacher Report"
            className="source-icon levitate"
            src="https://besticon-demo.herokuapp.com/icon?url=https://bleacherreport.com/&size=70..120..200"
            style={
              Object {
                "backgroundColor": "#000000",
              }
            }
          />
        </div>
        <div
          className="source-title"
        >
          Bleacher Report
        </div>
      </a>
      <a
        className="source animated fadeIn"
        href="/cnn"
        onClick={[Function]}
      >
        <div
          className="source-icon-container"
        >
          <img
            alt="CNN"
            className="source-icon levitate"
            src="https://besticon-demo.herokuapp.com/icon?url=https://www.cnn.com//&size=70..120..200"
            style={
              Object {
                "backgroundColor": "#cc0000",
              }
            }
          />
        </div>
        <div
          className="source-title"
        >
          CNN
        </div>
      </a>
      <a
        className="source animated fadeIn"
        href="/vice-news"
        onClick={[Function]}
      >
        <div
          className="source-icon-container"
        >
          <img
            alt="Vice News"
            className="source-icon levitate"
            src="https://besticon-demo.herokuapp.com/icon?url=https://news.vice.com/en_us/&size=70..120..200"
            style={
              Object {
                "backgroundColor": "#000000",
              }
            }
          />
        </div>
        <div
          className="source-title"
        >
          Vice News
        </div>
      </a>
      <a
        className="source animated fadeIn"
        href="/techcrunch"
        onClick={[Function]}
      >
        <div
          className="source-icon-container"
        >
          <img
            alt="TechCrunch"
            className="source-icon levitate"
            src="https://besticon-demo.herokuapp.com/icon?url=https://techcrunch.com//&size=70..120..200"
            style={
              Object {
                "backgroundColor": "#0D9307",
              }
            }
          />
        </div>
        <div
          className="source-title"
        >
          TechCrunch
        </div>
      </a>
      <a
        className="source animated fadeIn"
        href="/crypto-coins-news"
        onClick={[Function]}
      >
        <div
          className="source-icon-container"
        >
          <img
            alt="Crypto News"
            className="source-icon levitate"
            src="https://besticon-demo.herokuapp.com/icon?url=https://www.ccn.com//&size=70..120..200"
            style={
              Object {
                "backgroundColor": "#fea339",
              }
            }
          />
        </div>
        <div
          className="source-title"
        >
          Crypto News
        </div>
      </a>
    </div>
  </div>
</div>
`;
